src/lib/operators/join_sort_merge.cpp

```diff
@@ -139,6 +140,24 @@ class JoinSortMerge::JoinSortMergeImpl : public AbstractReadOnlyOperatorImpl {
   std::vector<std::shared_ptr<RowIDPosList>> _output_pos_lists_left;
   std::vector<std::shared_ptr<RowIDPosList>> _output_pos_lists_right;
 
+  struct RowHasher {
+    size_t operator()(const RowID& row) const {
+      auto seed = size_t{0};
+      boost::hash_combine(seed, row.chunk_id);
+      boost::hash_combine(seed, row.chunk_offset);
+      return seed;
+    }
+  };
+
+  using RowHashSet = std::unordered_set<RowID, RowHasher>;
+  // these are used for outer joins where the primary predicate is not Equals.
```

Comment:
Should be a line below and that comment is not fully correct, right? It's about Outer with secondary predicates, isn't it?

Reply:
Also please make sure to proper capitalize comments.